pkg/controllers/nodeclass/suite_test.go

```diff
@@ -821,6 +820,47 @@ var _ = Describe("NodeClassController", func() {
 		BeforeEach(func() {
 			profileName = instanceprofile.GetProfileName(ctx, fake.DefaultRegion, nodeClass)
 		})
+		It("should fail to delete the launch template", func() {
```

Comment:
An idea for an additional test: ensure multiple identical EC2NodeClasses create unique launch templates with the correct tags. Probably shouldn't be here in the nodeclass functional testing, maybe in the launchtemplate functional testing or in an E2E test.

Reply:
Thanks for the idea. I have added a test to the launchtemplate functional testing.